compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/StreamingCodeDesugar.java

```diff
@@ -0,0 +1,812 @@
+        BLangVariableDef outputStreamFunctionVarDef = ASTBuilderUtil.createVariableDef(lambdaFunction.pos,
+                outputStreamFunctionVariable);
+
+        stmts.add(outputStreamFunctionVarDef);
+
+        //Create output event process definition
+        BLangSimpleVarRef outputStreamFunctionSimpleVarRef = ASTBuilderUtil.createVariableRef(lambdaFunction.pos,
+                outputStreamFunctionVariable.symbol);
+        BInvokableSymbol outputProcessInvokableSymbol = (BInvokableSymbol) symResolver.
+                resolvePkgSymbol(lambdaFunction.pos, env, names.fromString(STREAMS_STDLIB_PACKAGE_NAME)).
+                scope.lookup(new Name("createOutputProcess")).symbol;
+
+        BType outputProcessInvokableType = outputProcessInvokableSymbol.type.getReturnType();
+        BVarSymbol outputProcessInvokableTypeVarSymbol = new BVarSymbol(0,
+                new Name(getVariableName(OUTPUT_PROCESS_FUNC_REFERENCE)), outputProcessInvokableSymbol.pkgID,
+                outputProcessInvokableType, env.scope.owner);
+        nextProcessVarSymbolStack.push(outputProcessInvokableTypeVarSymbol);
+
+        List<BLangExpression> args = new ArrayList<>();
+        args.add(outputStreamFunctionSimpleVarRef);
+        BLangInvocation outputProcessMethodInvocation = ASTBuilderUtil.
+                createInvocationExprForMethod(lambdaFunction.pos, outputProcessInvokableSymbol, args,
+                        symResolver);
+        outputProcessMethodInvocation.argExprs = args;
+        BLangVariable outputProcessInvokableTypeVariable = ASTBuilderUtil.
+                createVariable(lambdaFunction.pos, getVariableName(OUTPUT_PROCESS_FUNC_REFERENCE),
+                        outputProcessInvokableType, outputProcessMethodInvocation, outputProcessInvokableTypeVarSymbol);
+
+
+        BLangUserDefinedType userDefinedType = (BLangUserDefinedType) TreeBuilder.createUserDefinedTypeNode();
+        userDefinedType.typeName = ASTBuilderUtil.createIdentifier(lambdaFunction.pos, "OutputProcess");
+        userDefinedType.type = outputProcessInvokableType;
+        outputProcessInvokableTypeVariable.setTypeNode(userDefinedType);
+        BLangVariableDef outputProcessInvokableTypeVariableDef = ASTBuilderUtil.createVariableDef(lambdaFunction.pos,
+                outputProcessInvokableTypeVariable);
+
+        stmts.add(outputProcessInvokableTypeVariableDef);
+    }
+
+    @Override
+    public void visit(BLangSelectClause selectClause) {
+
+        //Create lambda function Variable
+        BLangVariable lambdaFunctionVariable =
+                this.createStreamEventTypeVariable(getVariableName(SIMPLE_SELECT_LAMBDA_PARAM_REFERENCE),
+                        selectClause.pos, env);
+
+        //Create new lambda function to process the output events
+        BLangLambdaFunction simpleSelectLambdaFunction = (BLangLambdaFunction) TreeBuilder.createLambdaFunctionNode();
```

Comment:
lambda functions are created in several places. May be its better to move that logic to a common method and re-use?

Reply:
Done. Made changes in possible places.